components/automate-ui/src/app/components/form-control/form-control.directive.ts

```diff
@@ -72,6 +72,13 @@ export class FormControlDirective implements OnInit, OnDestroy, OnChanges {
         distinctUntilChanged()
       )
       .subscribe(newValue => {
+        // If null, undefined, or empty string, we came in to early last time.
+        // So take this value as the originalValue.
+        if (!this.originalValue) {
+          this.originalValue = newValue;
+        }
```

Comment:
Should this  be an `else`, or a single conditional? When I read this, it looks like we first check for the original value. If it's not present, we set the original value to the new value . . . and then we immediately check if the new value is equal to the original value.

Reply:
Good question! They need to be separate conditionals. Here is why: say we come in with the first non-empty value after having come in earlier with an empty value. We need to first set the originalValue to this first non-empty value, then perform the normal check of seeing if the new value === original value. Of course, it will always be true at that point, so we do a `reset()`--which is vital to do at that point. But now consider values that come in later. We will no longer adjust the originalValue, but we will continue to check if the new value === original value.